### PR TITLE
Synchronize gwcs region with changes in jwst/romancal

### PR DESCRIPTION
**Pull request by WilliamJamieson**
There are copies of `region.py` in both jwst and romancal. This PR brings the gwcs version in line with these versions, so they can be factored out.

### CODE REVIEW

File: gwcs/region.py [hunk 1]
```diff
@@ -372,21 +389,25 @@ def intersection(self, edge):
...
-        D = _cross(u, v)
 
-        if abs(D) <= 1e2 * np.finfo(float).eps:
+        # Find the determinant of the matrix formed by the vectors u and v
+        #    Note: Originally this was computed using a numpy "2D" cross product,
+        #          however, this functionality has been deprecated and slated for
+        #          removal.
+        D = np.linalg.det([u, v])
+
+        if np.allclose(D, 0, rtol=0, atol=1e2 * np.finfo(float).eps):
```
[the remark this answers is not in the document]
Comment (WilliamJamieson) [reply gwcs/region.py:399]: I you are correct that your "`_cross`" method is faster at 2.56 ms vs 18.2 ms over 100,000 samples; however, both are faster than `np.cross` at 54.3 ms over the same input set. If you do want to go with the special method then I think it should be renamed to reflect the fact it is not really a cross product as the cross product does not make sense in 2D, but rather its really just a determinant of the matrix induced by the vectors.

**Review by mcara — APPROVED** [review]
LGTM. Thanks!

File: gwcs/region.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcara) [reply gwcs/region.py]: You are correct about `_cross` in 2D.

**Comment by mcara** [comment]
Do we really need `np.allclose` here? I think a simple `abs(det) < _INTERSECTION_ATOL` is simple and clear enough and without all the numpy array overhead:

```python
In [4]: %timeit np.allclose(x, 0, 0, 1e-6)
12.4 μs ± 15.9 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [5]: %timeit abs(x) < 1.e-6
23.6 ns ± 0.0665 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

**Comment by WilliamJamieson** [comment]
> Do we really need `np.allclose` here? I think a simple `abs(det) < _INTERSECTION_ATOL` is simple and clear enough and without all the numpy array overhead:
> 
> ```python
> In [2]: import numpy as np
> 
> In [3]: x = 1.e-12
> 
> In [4]: %timeit np.allclose(x, 0, 0, 1e-6)
> 12.4 μs ± 15.9 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
> 
> In [5]: %timeit abs(x) < 1.e-6
> 23.6 ns ± 0.0665 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
> ```

Sure

**Pull request merged**